### DISCUSSION TITLE
Engaging modes

### DISCUSSION THREAD

**Discussion by hillaryan** [opensteno/plover#1457] [Q&A and Support] [answered]
Suppose you want to type the following: `this is my_variable`.
My understanding is you would use the following sequence:
```
this is my {MODE:SNAKE} variable
```
Rather than this
```
this is {MODE:SNAKE} my variable
```
which yields `this is_my_variable`

This behavior makes sense, but it would be nice if there was a way that the later sequence could produce the former's output. Is there a way to achieve this?

**Comment by JoshuaGrams** [comment, marked as answer]
Probably the closest thing is a command which does `{^ ^}{mode:snake}` (so manually inserting a space and then switching to snake mode)?

Aerick also mentioned (on Discord) the [plover_auto_identifier](https://github.com/user202729/plover_auto_identifier) plugin as a possibility, though he hasn't used it (and neither have I).

**Reply by hillaryan** [reply]
Thank you for the quick response. I think that would work well except at the beginning of lines.

**Reply by user202729** [reply]
Alternatively use some plugin to retrospectively insert underscore (this requires multiple strokes, but since they're consecutive it should not be a large problem)

The problem doesn't sound very hard, just insert a "word" action with zero characters then add `{^}` and switch to snake mode, but it's not representable in the current Plover dictionary format. (can write a meta plugin?)

Regarding the auto identifier plugin, there are a few cases where it's a little annoying, I should fix them soon.
